rpc/json/ws.go

```diff
@@ -33,13 +33,40 @@ func (wsc *wsConn) sendLoop() {
 	}
 }
 
+func CheckOrigin(r *http.Request, allowedOrigins []string) bool {
+	origin := r.Header.Get("Origin")
+	if origin == "" {
+		return true
+	}
+
+	if len(allowedOrigins) > 0 && allowedOrigins[0] == "*" {
+		return true
+	}
+	for _, allowed := range allowedOrigins {
+		if allowed == origin {
+			return true
+		}
+	}
+	return false
+}
+
+func OriginChecker(allowedOrigins []string) func(*http.Request) bool {
+	return func(r *http.Request) bool {
+		return CheckOrigin(r, allowedOrigins)
+	}
+}
+
 func (h *handler) wsHandler(w http.ResponseWriter, r *http.Request) {
 	// TODO(tzdybal): configuration options
 	upgrader := websocket.Upgrader{
 		ReadBufferSize:  1024,
 		WriteBufferSize: 1024,
 	}
 
+	// TODO: use CORSAllowedMethods, CORSAllowedHeaders
+	if h.config != nil {
+		upgrader.CheckOrigin = OriginChecker(h.config.CORSAllowedOrigins)
```

Comment:
also, not need for many function calls and checks. you could inline something like (if it is sufficeint):
```
if h.config.IsCorsEnabled() {
	upgrader.CheckOrigin = func(r *http.Request) bool {
		for _, allowedOrigin := range h.config.CORSAllowedOrigins {
			if allowedOrigin == r.Header["Origin"][0] {
				return true
			}
		}
		return false
	}
}
```

Reply:
I'm a big fan of inlining here, instead of creating a factory method to create a function.